staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/dynamic_serving_content.go

```diff
@@ -54,7 +54,14 @@ var _ CertKeyContentProvider = &DynamicCertKeyPairContent{}
 var _ ControllerRunner = &DynamicCertKeyPairContent{}
 
 // NewDynamicServingContentFromFiles returns a dynamic CertKeyContentProvider based on a cert and key filename
+//
+//logcheck:context //NewDynamicServingContentFromFilesWithContext should be used instead because ret.loadCertKeyPair is called and works better when a context is supplied
 func NewDynamicServingContentFromFiles(purpose, certFile, keyFile string) (*DynamicCertKeyPairContent, error) {
+	return NewDynamicServingContentFromFilesWithContext(context.Background(), purpose, certFile, keyFile)
+}
+
+// NewDynamicServingContentFromFilesWithContext returns a dynamic CertKeyContentProvider based on a cert and key filename
+func NewDynamicServingContentFromFilesWithContext(ctx context.Context, purpose, certFile, keyFile string) (*DynamicCertKeyPairContent, error) {
```

Comment:
It is awkward that we use one logger during construction (`New`) and then a different logger in the background (`Run(ctx)`), but I don't see a way around that.

`logger klog.Logger` is a better parameter here.